modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/schema/SchemaHolderImpl.java

```diff
@@ -371,13 +377,29 @@ private static Object affinityIdentity(CacheConfiguration<?, ?> ccfg) {
         Class<?>[] colTypes,
         String[] colNames
     ) {
+        if (!checkNewUserDefinedFunction(schemaName, name))
+            return;
+
         IgniteSchema schema = igniteSchemas.computeIfAbsent(schemaName, IgniteSchema::new);
 
         schema.addFunction(name.toUpperCase(), IgniteTableFunction.create(method, colTypes, colNames));
 
         rebuild();
     }
 
+    /** */
+    private boolean checkNewUserDefinedFunction(String schName, String funName) {
+        if (F.eq(schName, QueryUtils.DFLT_SCHEMA)
+            && RexImpTable.INSTANCE.hasOperator(op -> op instanceof SqlFunction && op.getName().equalsIgnoreCase(funName))) {
```

Comment:
RexImpTable - it's not a catalog of all system functions, it's implementor table. Some functions are absent in RexImpTable (convertlet can transform some function to another, for example, there is no DECODE or NVL.. in RexImpTable). Right place to find system functions - is operator table (see frameworkCfg.getOperatorTable().lookupOperatorOverloads())